api-docs/api-reference/methods/roles/get-list-effective-roles-assigned-to-a-user-v2.0.rst

```diff
@@ -0,0 +1,400 @@
+   :widths: 30 20 30
+
+   * - Name
+     - Type
+     - Description
+   * - **RAX-AUTH:roleAssignments**
+     - Object *(Required)*
+     - A role assignments entity.
+   * - RAX-AUTH:roleAssignments.\ **tenantAssignments**
+     - Object *(Required)*
+     - An array of tenant assignments.
+   * - RAX-AUTH:roleAssignments.tenantAssignments.\ **onRole**
+     - String *(Required)*
+     - Role ID for tenant assignment.
+   * - RAX-AUTH:roleAssignments.tenantAssignments.\ **onRoleName**
+     - String *(Required)*
+     - Role name for tenant assignment.
+   * - RAX-AUTH:roleAssignments.tenantAssignments.\ **forTenants**
+     - String *(Required)*
+     - An array of tenant IDs to which the role is assigned. This is the union
+       of all tenants across all sources for the role
+   * - RAX-AUTH:roleAssignments.tenantAssignments.\ **sources**
+     - An array of sources *(Required)*
+     - An array of sources which grant the user the role on the specified
+       tenant(s)
+   * - RAX-AUTH:roleAssignments.tenantAssignments.sources.\ **sourceType**
+     - String *(Required)*
+     - The source of the role assignment. This can be
+        - USER
+        - USERGROUP
+        - SYSTEM
+   * - RAX-AUTH:roleAssignments.tenantAssignments.sources.\ **sourceId**
+     - String *(Required)*
+     - A unique identifier for the source. For users and groups this is the
+       respective id. For SYSTEM, this is the source system (e.g. IDENTITY)
+   * - RAX-AUTH:roleAssignments.tenantAssignments.sources.\ **assignmentType**
+     - String *(Required)*
+     - How tenants are determined for the source. This can be one of
+        - DOMAIN
+        - TENANT
+        - RCN
+   * - RAX-AUTH:roleAssignments.tenantAssignments.sources.\ **forTenants**
+     - Array of Strings *(Required)*
+     - The list of tenants to which the source grants the user the role.
+
+-------
+Sources
+-------
+The response includes one or more sources for each role that the user has
+assigned. Each source includes a source type, source id, assignment type, and
+tenants on which that source granted the user the specified role.
+
+SourceType and SourceId
+^^^^^^^^^^^^^^^^^^^^^^^
+A user can receive a given role through multiple methods.
+
+.. list-table:: Source Type
+   :header-rows: 1
+   :widths: 20 60
+
+   * - Type
+     - Description
+   * - USER
+     - Direct role assignment to the user
+   * - USERGROUP
+     - Indirect assignment to the user via user groups
+   * - SYSTEM
+     - Identity System level functionality that automatically assigns certain
+       roles
+
+The ``sourceId`` identifies the id of the ``sourceType``. For example, the
+``sourceId`` for the source with a ``USER`` source type is the ID for the
+user to which the role was directly assigned, whereas the ``sourceId`` for a
+``USERGROUP`` type would be the ID for the user group.
+
+Assignment type
+^^^^^^^^^^^^^^^
+Role assignments are classified in to three ways based on how the tenants for
```

Comment:
Role assignments are classified based on how the tenants, for which those assignments apply, are determined. The classifications are shown in the following table: